src/core/flow.ts

```diff
@@ -163,6 +163,10 @@ export function Flow({
       onEnd: HookWrapper(phaseConfig.turn.onEnd),
       endIf: TriggerWrapper(phaseConfig.turn.endIf),
     };
+
+    if (typeof phaseConfig.next === 'function') {
+      phaseConfig.wrapped.next = HookWrapper(phaseConfig.next);
+    }
```

Comment:
Could we always wrap `next` so that later code doesn’t need to check its type?

Something like:

```suggestion
if (typeof phaseConfig.next !== 'function') {
  const { next } = phaseConfig;
  phaseConfig.next = () => next;
}
phaseConfig.wrapped.next = HookWrapper(phaseConfig.next);
```

Reply:
Great approach. Wish I'd thought of it!